app/src/main/java/org/mozilla/focus/ScreenController.kt

```diff
@@ -94,4 +99,35 @@ object ScreenController {
         // Activates the start up code path, which creates a new session if there are none.
         SessionManager.getInstance().removeAllSessions()
     }
+
+    fun setNavigationOverlayIsVisible(
+        fragmentManager: FragmentManager,
+        appBarLayoutController: BrowserAppBarLayoutController,
+        isVisible: Boolean,
+        isOverlayOnStartup: Boolean
+    ) {
+        fun getNavOverlay() = fragmentManager.getNavigationOverlay()
+
+        if (isVisible) {
+            if (getNavOverlay() != null) {
+                throw IllegalStateException("Did not expect navigation overlay to exist")
+            }
```

Comment:
It seems odd to me that calling with `isVisible = true` twice in a row will throw an exception, but calling it twice in a row with `isVisible = false` won't.  I feel like it should either strictly enforce expected state or not.

Also, this would be a good thing to add in a kdoc.

Reply:
Calling false twice will throw: the fragment manager will return null for the navigation overlay and we `!!` it. I'll comment about this and document that this will throw in the kdoc.